### PR TITLE
update code paths in the `network` crate

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed
Propagates the changes to the rpc and gossip types up the network crate at a code path level: i.e. Logic is either updated when trivial, or commented out and added as `unimplemented!()`. This makes at least the `network` crate compile.

**NOTE** THIS IS A VERY UGLY AND GROSS PR

Depends on 
- [ ] #4064

### CODE REVIEW

File: beacon_node/network/src/beacon_processor/worker/gossip_methods.rs [hunk 1]
```diff
@@ -647,6 +648,33 @@ impl<T: BeaconChainTypes> Worker<T> {
         }
     }
 
+    /// Process the beacon block received from the gossip network and:
+    ///
+    /// - If it passes gossip propagation criteria, tell the network thread to forward it.
+    /// - Attempt to add it to the beacon chain, informing the sync thread if more blocks need to
+    ///   be downloaded.
+    ///
+    /// Raises a log if there are errors.
```
Comment (jimmygchen) [inline beacon_node/network/src/beacon_processor/worker/gossip_methods.rs]: just a reminder that the comment needs update here, possibly when it's implemented

File: beacon_node/network/src/beacon_processor/worker/rpc_methods.rs [hunk 2]
```diff
@@ -219,24 +220,49 @@ impl<T: BeaconChainTypes> Worker<T> {
...
+                                if (known_index as u64) == index {
+                                    let blob_sidecar = types::BlobSidecar{
+                                        block_root: beacon_block_root, 
+                                        index,
+                                        slot: beacon_block_slot,
+                                        block_parent_root: block.parent_root,
+                                        proposer_index: block.proposer_index,
+                                        blob,
+                                        kzg_commitment: block.body.blob_kzg_commitments[known_index].clone(), // TODO: needs to be stored in a more logical way so that this won't panic.
+                                        kzg_proof: kzg_aggregated_proof // TODO: yeah
```
Comment (jimmygchen) [inline beacon_node/network/src/beacon_processor/worker/rpc_methods.rs:256]: > TODO: yeah

lol

File: beacon_node/network/src/beacon_processor/worker/gossip_methods.rs (hunk 1 above)
Comment (divagant-martian) [reply beacon_node/network/src/beacon_processor/worker/gossip_methods.rs]: I'll remove it

**Review by pawanjay176 — APPROVED** [review]
I'm happy to merge this to reduce branch fatigue. Rebased this on `deneb-free-blobs` after merging in #4064

**Pull request merged**